### PR TITLE
User login

### LINKED ISSUE #8: Receive Login Information
As a user,
I can login with email and password,
where the app send information
```
{
  "email": "name@email.com",
  "password": "password"
}
``` 
to `POST /api/v1/sessions`,
which finds user in database,
and returns the response
```
status: 201
body:
{
  "data":
  {
    "type": "user",
    "id": "1",
    "attributes":
    {
      "first_name": "First Name",
      "last_name": "Last Name",
      "phone_number": "3033033030",
      "email": "FirstLast@email.com",
      "api_key": "asD7fsdfwef2332!%sdf"
    }
  }
}
```
### Checks
- [x] For success, returns status code 200
- [x] Returns correct response format json
- [x] Add endpoint documentation to README.md

Sad path
- [x] Unable to login, returns status code 422 with message: "Failed to login"

### PR DESCRIPTION
**Pull request by lpile**
## Changes proposed in this pull request:
* Add user login spec
* Add #create action to SessionsController with route
* Add endpoint documentation to README.md

## Card(s) closed in this pull request:
close #8

## What did you struggle on to complete?
* Nothing at this moment.

### Test Coverage Percentage: 100%
- [x] Tested my new feature(s) as well as any feasible edge cases (if possible)
- [x] Checked coverage/index.html - did not add any new code that's not covered by testing (if possible)
- [x] Ran the test suite - all tests are passing
- [x] Merged in the latest master to my branch with git pull origin master & resolved merge conflicts
- [x] I have commented my code, particularly in hard-to-understand areas

## Helpful Resources:
* [Authentication blog](https://medium.com/@wintermeyer/authentication-from-scratch-with-rails-5-2-92d8676f6836)

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -1,11 +1,19 @@
...
+      # All users
       resources :users, only: [:create]
+
+      # Specific User
       namespace :user do
         resources :events, only: [:index]
       end
+
+      # User Login
+      post '/sessions', to: 'sessions#create'
```
Comment (annekemcgrady) [inline config/routes.rb]: Logan can you run me through what is happening on line 16,  curious about the post here.
